core/src/main/scala/kafka/log/Log.scala

```diff
@@ -217,6 +218,9 @@ class Log(@volatile var dir: File,
         if (indexFileExists) {
           try {
             segment.index.sanityCheck()
+            // Resize the time index file to 0 if it is newly created.
+            if (!timeIndexFileExists)
+              segment.timeIndex.resize(0)
```

Comment:
@lindong28 @junrao Thanks for the review.

The index of the active segment was resized back to configured max index size later. 

The broker does not know which segment is the active segment until it fully loads all the segments. So if we want to handle the active segment separately, there are two options:

1. trim the time indices of all the segments here if they are newly created, and resize the active segment later after we know which segment is the active segment.

2. trim the time indices later after we know which segment is the active segment, we either need to pass the information of newly created time indices or we have to trim all the time indices to valid size.

I initially chose option 1 because it would be a one time cost during migration. If we go with option 2, we will have to pay the cost all the time unless we remember which indices are newly created which adds a little more code.

I updated the patch using option 2. The two options both seem fine to me. So please let me know which one do you prefer.